### ISSUE TITLE
Building on Debian 8

### ISSUE THREAD

**Issue by explorer85** [mattvchandler/textogl#1]
Hello! I'm having problems with building a library on Debian 8, with cmake version 3.0.2.
First I replaced project(textogl CXX)  with a project(textogl CXX C), because the following error occurred:
```
CMake Error at /usr/share/cmake-3.0/Modules/CheckLibraryExists.cmake:52 (try_compile):
  Unknown extension ".c" for file
```

Now I have successfully completed 
$ cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_SHARED_LIBS=1
 but I can not perform the $ make , because of the following error:
```
Scanning dependencies of target shaders
make[2]: *** No rule to make target 'shaders.inl', needed by 'src/CMakeFiles/shaders'.  Stop.
CMakeFiles/Makefile2:76: recipe for target 'src/CMakeFiles/shaders.dir/all' failed
Makefile:137: recipe for target 'all' failed
make[1]: *** [src/CMakeFiles/shaders.dir/all] Error 2
make: *** [all] Error 2
```

Please help me if possible, and sorry for my English:)

**Comment by mattvchandler** [comment]
Are you using this as part of a lager project, or trying to build the demo? If the latter, I have no idea why you would get those errors. There aren't any C files in the project.
If it is a larger project, would you share your CMakeLists.txt file(s)?

**Comment by explorer85** [comment]
I use this to build a demo.
I tried to collect it in Debian 9 with cmake version 3.7.2 and got success.
Perhaps the problem is CMAKE.
What is the minimum version of CMake supported by your library
Are you sure that cmake_minimum_required (VERSION 3.0) in your CMakeLists.txt is true?

**Comment by explorer85** [comment]
It seems to me that your textogl / CMakeLists.txt is not compatible with cmake version 3.0.2.

**Comment by mattvchandler** [comment]
Apparently it wasn't.
set(CMAKE_CXX_STANDARD 11) is in 3.1
and
add_custom_target(BYPRODUCTS is in 3.2

I've pulled these out and replaced them with 3.0 compatible commands

Should be fixed in 9bec17e896c6410ed64116d52cb3cdb3ab4ff933

There's also a few other bugs that popped up running it in Debian 8, so I fixed those as well in master

Let me know if that doesn't work.
